components/AuthModal/AuthModal.tsx

```diff
@@ -2,76 +2,80 @@ import { Dialog, Transition } from '@headlessui/react';
 import { Fragment } from 'react';
 import { useAppSelector, useAppDispatch } from '../../app/hooks';
 import { closeModal } from '../../store/slices/sliceModal';
-import { signInWithPopup } from 'firebase/auth';
+import { signInWithPopup, signOut } from 'firebase/auth';
 import { auth, Provider_github, Provider_google } from '../../lib/firebase';
 import toast, { Toaster } from 'react-hot-toast';
 import { IAuthData } from './Auth.interface';
 import 'twin.macro';
+import { setUserLoggedOut } from 'store/slices/sliceUser';
+
+enum ProviderType {
+  'GOOGLE',
+  'GITHUB',
+}
 
 export const AuthModal = ({ title }: IAuthData) => {
   const mode = useAppSelector((state) => state.mode.mode);
   const isOpen = useAppSelector((state) => state.modal.modal);
   const dispatch = useAppDispatch();
-  const handleGoogleSignin = () => {
-    signInWithPopup(auth, Provider_google)
-      .then(() => {
-        mode
-          ? toast.success('Login Successful', {
-              position: 'bottom-center',
-              duration: 2000,
-              style: {
-                borderRadius: '10px',
-                background: '#333',
-                color: '#fff',
-              },
-            })
-          : toast.success('Login Successful', { position: 'bottom-center' });
-      })
-      .catch((error) =>
-        mode
-          ? toast.error(error.message, {
-              position: 'bottom-center',
-              duration: 2000,
-              style: {
-                borderRadius: '10px',
-                background: '#333',
-                color: '#fff',
-              },
-            })
-          : toast.error(error.message, { position: 'bottom-center' })
-      );
-    dispatch(closeModal());
+
+  const createUser = async (
+    email: string | null,
+    firebaseUID: string | null
+  ): Promise<boolean> => {
+    const body = {
+      email,
+      firebaseUID,
+    };
+    const res = await fetch('/api/user', {
+      method: 'POST',
```

Comment:
it will create user everytime they logs in. could you add a check before adding it? 
like forexample check if the firebaseUID/email is already present in the db. if it is present don't create user again. if it's not there then create user.

Reply:
is it implemented?

Reply:
we don't need to show this message to the user. it should be internal only. no need to show any toast message

Reply:
yes

Reply:
> 

it wont show, u can try in preview  mode

Reply:
i did try to login twice and it added users twice in the db. could you also please check?

Reply:
Maybe it's because Prisma's latest changes were pushed to DB. I use my own local DB for testing

Reply:
its solved. thanks @leewan-09